### ISSUE TITLE
Ability to turn off auto ticket redemption and manually redeem winning tickets

### ISSUE THREAD

**Issue by yondonfu** [livepeer/go-livepeer#1757]
**Is your feature request related to a problem? Please describe.**

Winning tickets currently expire at the end of the next round. Due to high gas prices, a user may want to determine on their own when to redeem the ticket i.e. when gas prices are lower. May be useful to have the following options:

- Turn off auto ticket redemption
- Expose a webserver endpoint for redeeming 1 or many tickets. This endpoint could then be in a curl request or used in user defined custom scripts with conditions

Would be good to spec this out a bit more.

The auto redeem logic could also be updated to be a bit smarter with regard to current gas prices. For example, a suggestion from Discord:

> Paying 0.12 ETH fees for a 0.17 ETH ticket.... couldn't the winning tickets somehow be redeemed at a later time (like within 24h)? E.g. only redeem when fees <50% of the value (but redeem anyways before it expires)

I could also see this type of logic being implemented in a user defined custom script if there was an endpoint for redeeming tickets.

**Comment by yondonfu** [comment]
There's been some great discussion about this on Discord so recording some of it here:

vires-in-numeris on an example strategy for redeeming tickets:

> Hmm let's see... 
> 1) Calculate the average gas price per hour based on the previous 24h (e.g. 6am: 110, 7am: 130, 8am: 200...)
> 2) Add 20 gwei or so to the lowest value and set it to your max gas price.
>3) Caim if current gas price < max gas price, but claim anyways if expiring < 1h"

@dob on modules for these strategies:

> I'd love to see "modules" for these sorts of strategies, for O selection strategies. People could submit different open source code blocks that implement an interface, and then node operators could start their node with configurations to indicate which strategy they want to use.

The endpoint mentioned in the OP might be able to address both of these suggestions. Expose the endpoint and create some example scripts that people can use.

**Comment by leszko** [comment]
@yondonfu @ArcAster Do you think it's still worth to implement this one (since the gas price is that an issue anymore with Arbitrum)?

**Comment by yondonfu** [comment]
Agreed that this is less important right now although we can certainly re-visit. Let's icebox for now.
